When running 
```python
from open_mastr import Mastr
import os

os.environ["NUMBER_OF_PROCESSES"] = "6"
data_bulk = ["biomass", "combustion", "gsgk", "hydro"]

if __name__ == "__main__":
    db = Mastr()
    db.download(data=data_bulk, date="2025-10-01", keep_old_downloads=True)

```

I get the error:
```bash
2025-12-17 10:35:41,819 [ERROR] Error processing file 'AnlagenEegWasser.xml': 'Start tag expected, '<' not found, line 1, column 1 (<string>, line 1)'
2025-12-17 10:35:43,900 [ERROR] Error processing file 'AnlagenEegBiomasse.xml': 'Start tag expected, '<' not found, line 1, column 1 (<string>, line 1)'
2025-12-17 10:35:44,280 [ERROR] Error processing file 'EinheitenWasser.xml': 'Start tag expected, '<' not found, line 1, column 1 (<string>, line 1)'
2025-12-17 10:35:47,530 [ERROR] Error processing file 'AnlagenKwk.xml': 'Start tag expected, '<' not found, line 1, column 1 (<string>, line 1)'
2025-12-17 10:35:49,018 [ERROR] Error processing file 'EinheitenBiomasse.xml': 'Start tag expected, '<' not found, line 1, column 1 (<string>, line 1)'
2025-12-17 10:35:58,554 [ERROR] Error processing file 'EinheitenVerbrennung.xml': 'Start tag expected, '<' not found, line 1, column 1 (<string>, line 1)'
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,8 @@ and the versioning aims to respect [Semantic Versioning](http://semver.org/spec/
 ### Changed
 - Adapt introduction text in README and docs
   [#647](https://github.com/OpenEnergyPlatform/open-MaStR/pull/647)
+- Hotfix of date selection and incorporate new url structure
+  [#699](https://github.com/OpenEnergyPlatform/open-MaStR/pull/699)
 ### Removed
 
 

--- a/open_mastr/xml_download/utils_cleansing_bulk.py
+++ b/open_mastr/xml_download/utils_cleansing_bulk.py
@@ -5,6 +5,7 @@ from open_mastr.xml_download.colums_to_replace import (
     columns_replace_list,
 )
 from zipfile import ZipFile
+import io
 
 
 def cleanse_bulk_data(df: pd.DataFrame, zipped_xml_file_path: str) -> pd.DataFrame:
@@ -59,8 +60,11 @@ def create_katalogwerte_from_bulk_download(zipped_xml_file_path) -> dict:
     """Creates a dictionary from the id -> value mapping defined in the table
     katalogwerte from MaStR."""
     with ZipFile(zipped_xml_file_path, "r") as f:
-        data = f.read("Katalogwerte.xml")
-        df_katalogwerte = pd.read_xml(data, encoding="UTF-16", compression="zip")
+        with f.open("Katalogwerte.xml") as xml_bytes_io:
+            xml_bytes = xml_bytes_io.read()
+        xml_text = xml_bytes.decode("utf-16")
+        xml_io = io.StringIO(xml_text)
+        df_katalogwerte = pd.read_xml(xml_io)
     katalogwerte_array = np.array(df_katalogwerte[["Id", "Wert"]])
     katalogwerte = dict(
         (katalogwerte_array[n][0], katalogwerte_array[n][1])

